lib/ansible/template/__init__.py

```diff
@@ -726,6 +767,16 @@ def _finalize(self, thing):
         '''
         if USE_JINJA2_NATIVE:
             return thing
+
+        if _is_rolled(thing):
+            # Auto unroll a generator, so that users are not required to
+            # explicitly use ``|list`` to unroll
+            # This only affects the scenario where the final result of templating
+            # is a generator, and not where a filter creates a generator in the middle
+            # of a template. See ``_unroll_iterator`` for the other case. This is probably
+            # unncessary
+            return list(thing)
```

Comment:
I was thinking of doing this for native jinja too, either here in `_finalize` or in `ansible_native_concat`, to solve https://github.com/ansible/ansible/issues/59019.

Reply:
I could probably move this above the jinja2 native check in that case then.  I just figured I wouldn't touch native, but I can see a benefit.

Reply:
Done.